src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -31,6 +32,24 @@ impl ScalarFunc {
         let res = t.date_format(format_mask_str)?;
         Ok(Some(Cow::Owned(res.into_bytes())))
     }
+
+    #[inline]
+    pub fn eval_data<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, Time>>> {
+        let t = try_opt!(self.children[0].eval_time(ctx, row));
+        if t.invalid_zero() {
```

Comment:
Seems we need to check `is_zero` here according to https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L269

Reply:
Done